src/app/shared/search/search-filters/search-filter/search-facet-filter/search-facet-filter.component.ts

```diff
@@ -110,66 +112,15 @@ export class SearchFacetFilterComponent implements OnInit, OnDestroy {
     this.currentPage = this.getCurrentPage().pipe(distinctUntilChanged());
 
     this.searchOptions$ = this.searchConfigService.searchOptions;
-    this.subs.push(this.searchOptions$.subscribe(() => this.updateFilterValueList()));
-    const facetValues$ = observableCombineLatest(this.searchOptions$, this.currentPage).pipe(
-      map(([options, page]) => {
-        return { options, page };
-      }),
-      switchMap(({ options, page }) => {
-        return this.searchService.getFacetValuesFor(this.filterConfig, page, options)
-          .pipe(
-            getFirstSucceededRemoteData(),
-            map((results) => {
-                return {
-                  values: observableOf(results),
-                  page: page
-                };
-              }
-            )
-          );
+    this.subs.push(
+      this.searchOptions$.subscribe(() => this.updateFilterValueList()),
+      this.refreshFilters.asObservable().pipe(
+        filter((toRefresh: boolean) => toRefresh),
+      ).subscribe(() => {
+        this.retrieveFilterValues(false);
```

Comment:
`retrieveFilterValues` shouldn't contain `subscribe`s if you need to use it here.

Reply:
done